examples/net/mqttclient/natiu/main.go

```diff
@@ -0,0 +1,135 @@
+// This example is an MQTT client built with the natiu-mqtt package.  It sends
+// machine.CPUFrequency() readings to the broker every second for 10 seconds.
+//
+// Note: It may be necessary to increase the stack size when using
+// paho.mqtt.golang.  Use the -stack-size=4KB command line option.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"machine"
+	"math/rand"
+	"net"
+	"time"
+
+	mqtt "github.com/soypat/natiu-mqtt"
+)
+
+var (
+	ssid   string
+	pass   string
+	broker string = "test.mosquitto.org:1883"
+	topic  string = "cpu/freq"
+)
```

Comment:
What are thoughts on defining `netdev` here as an interface and setting it in `init` functions in their respective board-specific files?

Reply:
Not sure I follow the question.  Can you work up an example?

Reply:
changing main.go to:
```go
var (
	ssid   string
	pass   string
	broker string = "test.mosquitto.org:1883"
	topic  string = "cpu/freq"
	netdev netdev.Netdev // Do we have this interface defined anywhere in drivers? 
)
```

Putting `netdev` declaration inside `init` function as an assignment (espat.go as an example)
```go
func init() {
    netdev = espat.New(&cfg)
}
```

The above code depends on having the Netdev interface defined somwhere.

Reply:
I tried this:

```
diff --git a/examples/net/tcpclient/main.go b/examples/net/tcpclient/main.go
index 76fd562..63b3d92 100644
--- a/examples/net/tcpclient/main.go
+++ b/examples/net/tcpclient/main.go
@@ -14,12 +14,15 @@ import (
        "machine"
        "net"
        "time"
+
+       "tinygo.org/x/drivers"
 )
 
 var (
        ssid string
        pass string
        addr string = "10.0.0.100:8080"
+       netdev drivers.netdever
 )
 
 var buf = &bytes.Buffer{}
diff --git a/examples/net/tcpclient/rtl8720dn.go b/examples/net/tcpclient/rtl8720dn.go
index 94d5277..fe6b823 100644
--- a/examples/net/tcpclient/rtl8720dn.go
+++ b/examples/net/tcpclient/rtl8720dn.go
@@ -26,4 +26,6 @@ var cfg = rtl8720dn.Config{
        WatchdogTimeout: time.Duration(20 * time.Second),
 }
 
-var netdev = rtl8720dn.New(&cfg)
+func init() {
+       netdev = rtl8720dn.New(&cfg)
+}
```

But it's a no-go because netdever isn't exported by drivers package:

```
+ tinygo flash -monitor -target wioterminal -stack-size=4KB -size short -ldflags '-X "main.ssid=test" -X "main.pass=testtest"' /home/sfeldma/work/tinygo-drivers/examples/net/tcpclient/
# tinygo.org/x/drivers/examples/net/tcpclient
examples/net/tcpclient/main.go:25:17: netdever not exported by package drivers
examples/net/tcpclient/main.go:34:19: netdev.NetConnect undefined (type drivers.netdever has no field or method NetConnect)
```